lib/bolt/transport/local.rb

```diff
@@ -146,25 +148,32 @@ def run_task(target, task, arguments, _options = {})
               environment_params = ""
             end
 
-            output =
-              if Powershell.powershell_file?(script) && stdin.nil?
-                command = Powershell.run_ps_task(arguments, script, input_method)
-                command = environment_params + Powershell.shell_init + command
+            if Powershell.powershell_file?(script) && stdin.nil?
+              command = Powershell.run_ps_task(arguments, script, input_method)
+              command = environment_params + Powershell.shell_init + command
+              interpreter ||= 'powershell.exe'
+              output =
                 if input_method == 'powershell'
-                  @conn.execute(command, dir: dir, env: "powershell.exe")
+                  @conn.execute(command, dir: dir, interpreter: interpreter)
                 else
-                  @conn.execute(command, dir: dir, stdin: stdin, env: "powershell.exe")
+                  @conn.execute(command, dir: dir, stdin: stdin, interpreter: interpreter)
                 end
+            end
+            unless output
```

Comment:
In general this nested logic for dealing with windows, interpreter, etc is getting messy. I got it working to develop tests. I think it could use some re-organization.